core/src/trezor/ui/layouts/tt/__init__.py

```diff
@@ -162,6 +162,16 @@ def confirm_multisig_warning() -> Awaitable[None]:
     )
 
 
+if not utils.BITCOIN_ONLY:
+
+    def confirm_unknown_contract_warning() -> Awaitable[None]:
```

Comment:
...this is _also_ strongly tied to Ethereum and should _also_ go to `ethereum/layout.py`, but the impl is different for mercury. could we maybe extract a building block from this? or use `show_warning` in all cases?

Reply:
Well, I could actually implement `warning_hi_prio` for all models and make it use the normal warning for non-Mercury. But the other issue with this is that the copy is also different for different models (changed it in the next PR). Non-Mercury have a shorter version of the text. How could I make that work without different implementations?

Reply:
yeah, that's the big copy problem. okay, leave it as-is, just please put "ethereum" in the name so that it's clearer.